pkg/velero/server/args.go

```diff
@@ -4,156 +4,278 @@ import (
-// +kubebuilder:object:generate=true
-type Args struct {
-	ServerConfig `json:",inline"`
-	GlobalFlags  `json:",inline"`
-}
+const (
+	Server = "server"
+
+	defaultFsBackupTimeout = 4 * time.Hour
 
-// GlobalFlags are flags that are defined across Velero CLI commands
-// +kubebuilder:object:generate=true
-type GlobalFlags struct {
-	client.VeleroConfig `json:",inline"`
-	klog.LoggingT       `json:",inline"`
+	TrueVal  = "true"
+	FalseVal = "false"
+)
+
+var defaultRestoreResourcePriorities = restore.Priorities{
+	HighPriorities: []string{
+		"securitycontextconstraints",
+		"customresourcedefinitions",
+		"namespaces",
+		"roles",
+		"rolebindings",
+		"clusterrolebindings",
+		"managedcluster.cluster.open-cluster-management.io",
+		"managedcluster.clusterview.open-cluster-management.io",
+		"klusterletaddonconfig.agent.open-cluster-management.io",
+		"managedclusteraddon.addon.open-cluster-management.io",
+		"storageclasses",
+		"volumesnapshotclass.snapshot.storage.k8s.io",
+		"volumesnapshotcontents.snapshot.storage.k8s.io",
+		"volumesnapshots.snapshot.storage.k8s.io",
+		"datauploads.velero.io",
+		"persistentvolumes",
+		"persistentvolumeclaims",
+		"serviceaccounts",
+		"secrets",
+		"configmaps",
+		"limitranges",
+		"pods",
+		"replicasets.apps",
+		"clusterclasses.cluster.x-k8s.io",
+		"endpoints",
+		"services",
+	},
+	LowPriorities: []string{
+		"clusterbootstraps.run.tanzu.vmware.com",
+		"clusters.cluster.x-k8s.io",
+		"clusterresourcesets.addons.cluster.x-k8s.io",
+	},
 }
 
-// StringArr returns the Velero server arguments as a string array
-// dpaFeatureFlags are the feature flags that are defined in the DPA CR which will be merged with Args
-// Most validations are done in the DPA CRD except float32 validation
-func (a Args) StringArr(dpaFeatureFlags []string, logLevel string) ([]string, error) {
-	args := []string{"server"}
-	// we are overriding args, so recreate args from scratch
-	if len(dpaFeatureFlags) > 0 {
-		args = append(args, fmt.Sprintf("--features=%s", strings.Join(dpaFeatureFlags, ",")))
-	}
-	if boolptr.IsSetToTrue(a.DefaultVolumesToFsBackup) {
-		args = append(args, "--default-volumes-to-fs-backup=true")
-	} else if boolptr.IsSetToFalse(a.DefaultVolumesToFsBackup) {
-		args = append(args, "--default-volumes-to-fs-backup=false")
-	}
-	if logLevel != "" {
-		logrusLevel, err := logrus.ParseLevel(logLevel)
-		if err != nil {
-			return []string{}, fmt.Errorf("invalid log level %s, use: %s", logLevel, "trace, debug, info, warning, error, fatal, or panic")
+func getFsBackupTimeout(dpa *oadpv1alpha1.DataProtectionApplication) string {
+	if dpa.Spec.Configuration.Restic != nil && len(dpa.Spec.Configuration.Restic.Timeout) > 0 {
+		fsBackupTimeout, err := time.ParseDuration(dpa.Spec.Configuration.Restic.Timeout)
+		if err == nil {
+			return fsBackupTimeout.String()
 		}
-		args = append(args, fmt.Sprintf("--log-level=%s", logrusLevel.String()))
+		// TODO should not error out?
```

Comment:
if user puts wrong string in restic or node agent timeout, code uses default and does not tell nothing to user